src/main/groovy/edu/oregonstate/mist/osuevents/db/CacheDAO.groovy

```diff
@@ -87,23 +91,33 @@ class CacheDAO {
         places
     }
 
+    /**
+     * Use as an intermediary method for vendor API endpoints that require pagination.
+     * Iterates through results until all pages have been requested and collected.
+     */
     private def pageIteration(String resource) {
         def data = []
-        Integer page = 1
-        def query = [pp:100]
-        query['page'] = page
+        def query = [pp:100, page:1]
 
-        data.add(page - 1,
+        data.add(query['page'] - 1,
```

Comment:
How about if you move this into a separate method or closure? You have three level deep of method calls and this is repeated in this method.